packages/node/src/utils.ts

```diff
@@ -73,9 +75,69 @@ export enum EdgeRuntimes {
   ExperimentalEdge = 'experimental-edge',
 }
 
+export enum NodejsRuntimes {
+  // @ts-ignore typescript does not allow undefined as a value
+  Legacy = undefined,
+  WinterCG = 'nodejs-experimental', // TODO remove -experimental once ready
+}
+
+export const ALLOWED_RUNTIMES = [
+  NodejsRuntimes.WinterCG as string,
+  ...Object.values(EdgeRuntimes),
+];
+
+export function checkConfiguredRuntime(
+  runtime: string | undefined,
+  entrypoint: string
+) {
+  if (runtime) {
+    // TODO remove when nodejs-experimental will land
+    if (runtime === 'nodejs') {
+      throw new Error(
+        `${entrypoint}: \`config.runtime: "nodejs"\` semantics will evolve soon. Please remove the \`runtime\` key to keep the existing behavior.`
+      );
+    }
+    if (!ALLOWED_RUNTIMES.includes(runtime)) {
+      throw new Error(
+        `${entrypoint}: unsupported "runtime" property in \`config\`: ${JSON.stringify(
+          runtime
+        )} (must be one of: ${JSON.stringify(
+          ALLOWED_RUNTIMES
+        )}). Learn more: https://vercel.link/creating-edge-functions`
```

Comment:
**question (non-blocking):** Do we even want to say that `experimental-edge` is a valid value here? Maybe we hide that from this list when it's displayed.